scripts/update_collection_db.py

```diff
@@ -0,0 +1,157 @@
+                yield owner, versioned_uuid
+
+            except AssertionError:
+                yield owner, versioned_uuid
+
+    def _collections_in_bucket_but_not_in_database(self):
+        """
+        Returns any (owner, versioned_uuid) present in the bucket but not in the collections table.
+
+        Returns an iterable tuple of strings: (owner, versioned_uuid) representing the item's key pair.
+
+        The returned keys can then be added to the collections dynamodb table.
+        """
+        for collection_key in self.handle.list(self.bucket, prefix=COLLECTION_PREFIX):
+            self.total_bucket_collection_items += 1
+            if collection_key != f'{COLLECTION_PREFIX}/':
+                uuid, version = collection_key[len(f'{COLLECTION_PREFIX}/'):].split('.', 1)
+                versioned_uuid = f'{uuid}.{version}'
+                if not self._is_uuid_tombstoned(uuid):
+                    try:
+                        collection = json.loads(self.handle.get(self.bucket, collection_key))
+                        if not owner_lookup.get_collection(owner=collection['owner'], versioned_uuid=versioned_uuid):
+                            yield collection['owner'], versioned_uuid
+                    except BlobNotFoundError:
+                        pass  # if deleted from bucket while being listed
+                    except KeyError:
+                        pass  # unexpected response
+                else:
+                    self.total_tombstoned_bucket_collection_items += 1
+
+    def remove_collections_from_database(self, all: bool=False):
+        if all:
+            collections = owner_lookup.get_all_collection_keys()
+            text = 'ALL'
+        else:
+            collections = self._collections_in_database_but_not_in_bucket()
+            text = 'INVALID'
+
+        print(f'\nRemoving {text} user-collection associations from database: {owner_lookup.collection_db_table}\n')
+        removed = 0
+        for owner, versioned_uuid in collections:
+            print(f"Removing {owner}'s collection: {versioned_uuid}")
+            owner_lookup.delete_collection(owner=owner, versioned_uuid=versioned_uuid)
+            removed += 1
+        print(f'{removed} collection items removed from: {owner_lookup.collection_db_table}')
+        return removed
+
+    def add_missing_collections_to_database(self):
+        print(f'\nAdding missing user-collection associations to database from: {self.bucket_name}\n')
+        added = 0
+        for owner, versioned_uuid in self._collections_in_bucket_but_not_in_database():
+            print(f"Adding {owner}'s collection: {versioned_uuid}")
+            owner_lookup.put_collection(owner=owner, versioned_uuid=versioned_uuid)
+            added += 1
+        print(f'{added} collection items added from: {self.bucket_name}')
+        return added
+
+
+def main():
+    start = time.time()
+    c = CollectionDatabaseTools(replica='aws')
+
+    if len(sys.argv) > 1:
```

Comment:
please use `argparse`

see https://github.com/HumanCellAtlas/data-store/blob/4e0a61c388013bb0499777a573c43ada7dd386ee/scripts/populate_lambda_ssm_parameters.py#L86

Reply:
I usually only use argparse if there's more than one option, like for this one that I did: https://github.com/HumanCellAtlas/data-store/blob/master/scripts/swagger_auth.py#L160

This is more based on browsing the scripts to determine the conventions in current script usage.  I'll start including argparse in all scripts then.